src/server/utils/replication_utils.js

```diff
@@ -0,0 +1,127 @@
+        if (cur_bucket.replication_policy_id &&
+            (cur_bucket.replication_policy_id.toString() === replication_id.toString())) {
+            acc.src_bucket = cur_bucket;
+        }
+        return acc;
+    }, { src_bucket: undefined, dst_bucket: undefined });
+
+    return ans;
+}
+
+function get_copy_type() {
+    // TODO: get copy type by src and dst buckets (for server side/other optimization)
+    return 'MIX';
+}
+
+async function move_objects(scanner_sem, client, copy_type, src_bucket_name, dst_bucket_name, keys) {
+    try {
+        const res = await scanner_sem.surround_count(keys.length,
+            async () => {
+                try {
+                    const res1 = await client.replication.move_objects_by_type({
+                        copy_type,
+                        src_bucket_name,
+                        dst_bucket_name,
+                        keys
+                    }, {
+                        auth_token: auth_server.make_auth_token({
+                            system_id: system_store.data.systems[0]._id,
+                            account_id: system_store.data.systems[0].owner._id,
+                            role: 'admin'
+                        })
+                    });
+                    return res1;
+                } catch (err) {
+                    // no need to do retries, eventually the object will be uploaded
+                    // TODO: serious error codes with metrics (auth_failed, storage not exist etc)
+                    dbg.error('replication_utils move_objects: error: ', err, src_bucket_name, dst_bucket_name, keys);
+                }
+            });
+        return res;
+    } catch (err) {
+        dbg.error('replication_utils move_objects: semaphore error:', err, err.stack);
+        // no need to handle semaphore errors, eventually the object will be uploaded
+    }
+}
+
+async function delete_objects(scanner_sem, client, bucket_name, keys) {
+    try {
+        const res = await scanner_sem.surround_count(keys.length,
+            async () => {
+                try {
+                    const res1 = await client.replication.delete_objects({
+                        bucket_name,
+                        keys
+                    }, {
+                        auth_token: auth_server.make_auth_token({
+                            system_id: system_store.data.systems[0]._id,
+                            account_id: system_store.data.systems[0].owner._id,
+                            role: 'admin'
+                        })
+                    });
+                    return res1;
+                } catch (err) {
+                    dbg.error('replication_utils delete_objects: error: ', err, bucket_name, keys);
+                }
+            });
+        return res;
+    } catch (err) {
+        dbg.error('replication_utils delete_objects: semaphore error:', err, err.stack);
+    }
+}
+
```

Comment:
According to your style before - add `// EXPORTS`.

Reply:
Marking resolved - addressed.